Neos.ContentRepository.Core/Classes/Projection/ContentGraph/NodeAggregate.php

```diff
@@ -182,14 +182,36 @@ public function getOccupationByCovered(DimensionSpacePoint $coveredDimensionSpac
     }
 
     /**
-     * Returns the dimension space points this aggregate is *explicitly* tagged in with the specified $subtreeTag
-     * NOTE: This won't respect inherited subtree tags!
+     * Get the dimensions this node aggregate is tagged according to the filter function
      *
-     * @internal This is a low level concept that is not meant to be used outside the core or tests
+     * Implementation note:
+     *
+     * We need to pass $nodeTagsByCoveredDimensionSpacePoint additionally to the NodeAggregate as this information doesn't exist otherwise.
+     * The nodes in $nodesByCoveredDimensionSpacePoint only point to the occupying nodes without entries for the specialisations.
+     * This means we CANNOT substitute this implementation by iterating over the occupied nodes as explicitly tagged specialisations will not show up as tagged:
+     *
+     *     foreach ($this->coveredDimensionSpacePoints as $coveredDimensionSpacePoint) {
+     *         $node = $this->nodesByCoveredDimensionSpacePoint[$coveredDimensionSpacePoint->hash];
+     *         if ($nodeTagFilter($node->tags)) {
+     *             $explicitlyTagged[] = $coveredDimensionSpacePoint;
+     *         }
+     *     }
+     *
+     * We could simplify this logic if we also add these specialisation node rows explicitly to the NodeAggregate.
+     *
+     * @param \Closure(NodeTags):bool $nodeTagFilter
+     * @internal Experimental api, this is a low level concept that is mostly not meant to be used outside the core or tests
      */
-    public function getDimensionSpacePointsTaggedWith(SubtreeTag $subtreeTag): DimensionSpacePointSet
+    public function filterCoveredDimensionsByNodeTags(\Closure $nodeTagFilter): DimensionSpacePointSet
```

Comment:
`filterCoveredDimensionsByTag(SubtreeTag::disabled(), withoutInherited: true)`

Reply:
+1 that would make it much more explicit and easier to use